tests/docker.etest

```diff
@@ -1001,3 +1000,160 @@ ETEST_docker_export_squashfs()
+    assert_emock_called_with "docker" 0 run     \
+        --mount "type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock"
+}
+
+ETEST_docker_run_copy_to_volume()
+{
+    emock "docker"
+
+    etestmsg "Calling docker_run with multiple copy_to_volume"
+
+    # Run this inside a subshell so that we can ensure the traps are getting called to remove the ephemeral volumes
+    (
+        docker_run --interactive=no \
+            --copy-to-volume "foo:/foo"                  \
+            --copy-to-volume "/home/marshall/data:/data" \
+            --copy-to-volume "/home/ebash:/baggins"      \
+    )
+
+    etestmsg "Verifying mock called 13 times"
+    assert_emock_called "docker" 13
+
+    etestmsg "Validating arguments (0) -- container create"
+    assert_emock_called_with "docker" 0 \
+        container create        \
+        --name foo              \
+        -v foo:/foo             \
+        busybox
+
+    etestmsg "Validating args (1) -- docker cp"
+    assert_emock_called_with "docker" 1 \
+        cp "foo/."              \
+        "foo:/foo"
+
+    etestmsg "Validating arguments (2) -- container create"
+    assert_emock_called_with "docker" 2 \
+        container create        \
+        --name data             \
+        -v data:/data           \
+        busybox
+
+    etestmsg "Validating args (3) -- docker cp"
+    assert_emock_called_with "docker" 3 \
+        cp "/home/marshall/data/."      \
+        "data:/data"
+
+    etestmsg "Validating arguments (4) -- container create"
+    assert_emock_called_with "docker" 4 \
+        container create        \
+        --name ebash            \
+        -v ebash:/baggins       \
+        busybox
+
+    etestmsg "Validating args (5) -- docker cp"
+    assert_emock_called_with "docker" 5 \
+        cp "/home/ebash/."      \
+        "ebash:/baggins"
+
+    etestmsg "Validating args (6) -- docker run"
+    assert_emock_called_with "docker" 6 \
+        run                     \
+        --volume=foo:/foo       \
+        --volume=data:/data     \
+        --volume=ebash:/baggins \
+
+    etestmsg "Validating args (7) -- docker rm"
+    assert_emock_called_with "docker" 7 \
+        rm ebash
+
+    etestmsg "Validating args (8) -- docker volume rm"
+    assert_emock_called_with "docker" 8 \
+        volume rm ebash
+
+    etestmsg "Validating args (9) -- docker rm"
+    assert_emock_called_with "docker" 9 \
+        rm data
+
+    etestmsg "Validating args (10) -- docker volume rm"
+    assert_emock_called_with "docker" 10 \
+        volume rm data
+
+    etestmsg "Validating args (11) -- docker rm"
+    assert_emock_called_with "docker" 11 \
+        rm foo
+
+    etestmsg "Validating args (12) -- docker volume rm"
+    assert_emock_called_with "docker" 12 \
+        volume rm foo
+
+}
+
+ETEST_docker_run_passthrough_args()
+{
+    emock "docker"
+
+    etestmsg "Calling docker_run with passthrough args"
+
+    docker_run --interactive=no -- \
```

Comment:
suggest to do an option that --interactive=yes?

Reply:
That's a good suggestion @bowanggu . I'll add that before I merge it.